bytecomp/bytelink.ml

```diff
@@ -656,8 +674,13 @@ let link objfiles output_name =
     Misc.try_finally
       ~always:(fun () -> List.iter remove_file !temps)
       (fun () ->
-         link_bytecode_as_c tolink c_file;
-         if not (Filename.check_suffix output_name ".c") then begin
+         let with_main = Filename.check_suffix output_name ".exe" in
```

Comment:
Note that the `.exe` extension is required in the argument to `-o` even on Unix.

Reply:
Indeed, I guess it might be slightly surprising for pure unix users, but at least it's in line with the current behaviour of `-output-complete-obj`. If that was really an issue in pratice, we could add a way to specify the output format instead of detecting it via the extension.